datafusion-federation/src/optimize.rs

```diff
@@ -1,27 +1,61 @@
 use datafusion::{
-    common::tree_node::{Transformed, TreeNode, TreeNodeRewriter},
+    common::tree_node::{Transformed, TransformedResult, TreeNode, TreeNodeRewriter},
     error::Result,
-    logical_expr:: LogicalPlan,
+    execution::{SessionState, SessionStateBuilder},
+    logical_expr::LogicalPlan,
     optimizer::{
-        optimizer::ApplyOrder, push_down_filter, OptimizerConfig, OptimizerContext, OptimizerRule
-    }
+        optimize_projections::OptimizeProjections, optimizer::ApplyOrder,
+        push_down_filter::PushDownFilter, OptimizerConfig, OptimizerRule,
+    },
+    prelude::SessionConfig,
 };
 
-pub(crate) fn optimize_plan(plan: LogicalPlan) -> Result<LogicalPlan> {
-    let push_down_filter_rule = push_down_filter::PushDownFilter::new();
-    // `push_down_filter` does not use config so it can be default
-    let optimizer_config = OptimizerContext::default();
+pub(crate) struct Optimizer {
```

Comment:
Wrapped in a struct to initialize requirements for optimization functionality only once (configuration + optimization rules)